File: typescript/src/tools/insights/delete.ts
import { InsightDeleteSchema } from "@/schema/tool-inputs";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = InsightDeleteSchema;

type Params = z.infer<typeof schema>;

export const deleteHandler = async (context: Context, params: Params) => {
	const { insightId } = params;
	const projectId = await context.getProjectId();
	const result = await context.api.insights({ projectId }).delete({ insightId });

	if (!result.success) {
		throw new Error(`Failed to delete insight: ${result.error.message}`);
	}

	return { content: [{ type: "text", text: JSON.stringify(result.data) }] };
};

const definition = getToolDefinition("insight-delete");

const tool = (): Tool<typeof schema> => ({
	name: "insight-delete",
	description: definition.description,
	schema,
	handler: deleteHandler,
});

export default tool;

File: python/tools/insights/delete.py
import json

from api.client import is_error, is_success
from schema.tool_inputs import InsightDeleteSchema
from tools.tool_definitions import get_tool_definition
from tools.types import Context, Tool, ToolResult


async def delete_insight_handler(context: Context, params: InsightDeleteSchema) -> ToolResult:
    project_id = await context.get_project_id()
    result = await context.api.insights(project_id).delete(int(params.insightId))

    if is_error(result):
        raise Exception(f"Failed to delete insight: {result.error}")

    assert is_success(result)
    return ToolResult(content=json.dumps(result.data))


def delete_insight_tool() -> Tool[InsightDeleteSchema]:
    definition = get_tool_definition("insight-delete")

    return Tool(
        name="insight-delete",
        description=definition["description"],
        schema=InsightDeleteSchema,
        handler=delete_insight_handler,
    )
